asdf.elv

```diff
@@ -87,20 +87,20 @@ for path [
   $asdf_data_dir'/shims'
 ] {
   if (not (has-value $paths $path)) {
-    paths = [
+    set paths = [
       $@paths
       $path
     ]
   }
 }
 
 # Setup argument completions
-fn arg-completer [@argz]{
-  argz = $argz[1:-1]  # strip 'asdf' and trailing empty string
+fn arg-completer {|@argz|
+  set argz = $argz[1:-1]  # strip 'asdf' and trailing empty string
```

Comment:
```suggestion
  set argz = $argz[1..-1]  # strip 'asdf' and trailing empty string
```
One thing the Elvish v0.17.0 upgrade script doesn't pick up on is the slice syntax. The legacy `a:b` syntax doesn't work anymore.

The same legacy slice syntax appears in lines 193 and 199.

Reply:
Done! Thanks